gtsfm/evaluation/visualize_benchmark_comparison.py

```diff
@@ -0,0 +1,254 @@
+"""
+Use Plotly to create a heatmap representing a 2d table, and add annotations to it.
+
+Authors: John Lambert
+"""
+
+import argparse
+from collections import defaultdict
+from typing import List
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import plotly
+import plotly.graph_objects as go
+from matplotlib import colors
+from matplotlib.colors import LinearSegmentedColormap
+from plotly.graph_objs.layout import Font, Margin, XAxis, YAxis
+
+import gtsfm.evaluation.merge_reports as report_utils
+import gtsfm.evaluation.metrics_report as metrics_report
+
+
+HEATMAP_WIDTH = 1500
+HEATMAP_HEIGHT = 900
+NUM_COLORS_COLORMAP = 100
+EPSILON = 1e-12
+MAX_NUM_CHARS_ARTIFACT_FNAME = 35
+
+MIN_RENDERABLE_PERCENT_CHANGE = -20
+MAX_RENDERABLE_PERCENT_CHANGE = 20
+
+ZIP_FNAMES = [
+    "deep_front_end-2011205_rc3-20-png-wget-astronet-1024-true.zip",
+    "deep_front_end-door-12-12-JPG-test_data-olsson-loader-1296-true.zip",
+    "deep_front_end-notre-dame-20-20-jpg-gdrive-colmap-loader-760-false.zip",
+    "deep_front_end-skydio-8-8-jpg-gdrive-colmap-loader-760-true.zip",
+    "deep_front_end-skydio-32-32-jpg-gdrive-colmap-loader-760-true.zip",
+    "sift_front_end-2011205_rc3-65-png-wget-astronet-1024-true.zip",
+    "sift_front_end-door-12-12-JPG-test_data-olsson-loader-1296-true.zip",
+    "sift_front_end-palace-fine-arts-281-25-jpg-wget-olsson-loader-320-true.zip",
+    "sift_front_end-skydio-8-8-jpg-gdrive-colmap-loader-760-true.zip",
+    "sift_front_end-skydio-32-32-jpg-gdrive-colmap-loader-760-true.zip",
+]
+
+TABLE_NAMES = [
+    "Verifier Summary",
+    "Inlier Support Processor Summary",
+    "Rotation Cycle Consistency Metrics",
+    "Cycle Consistent Frontend Summary",
+    "Averaging Metrics",
+    "Data Association Metrics",
+    "Bundle Adjustment Metrics",
+]
+
+RED_HEX = "#df0101"
+PALE_YELLOW_HEX = "#f5f6ce"
+GREEN_HEX = "#31b404"
+
+
+def colormap_to_colorscale(cmap):
+    # function that transforms a matplotlib colormap to a Plotly colorscale
+    return [colors.rgb2hex(cmap(k * 1 / NUM_COLORS_COLORMAP)) for k in range(NUM_COLORS_COLORMAP + 1)]
+
+
+def colorscale_from_list(alist, name):
+    # Defines a colormap, and the corresponding Plotly colorscale from the list alist
+    # alist=the list of basic colors
+    # name is the name of the corresponding matplotlib colormap
+
+    cmap = LinearSegmentedColormap.from_list(name, alist)
+    colorscale = colormap_to_colorscale(cmap)
+    return cmap, colorscale
+
+
+def plot_colored_table(X: List[str], Y: List[str], Z: np.ndarray, table_name: str) -> None:
```

Comment:
Looks like the returned type is not correct? Also please update documentation.

Reply:
good catch, i'll update